ncm-grub/src/main/pan/components/grub/schema.pan

```diff
@@ -7,6 +7,29 @@ declaration template components/grub/schema;
 
 include {'quattor/schema'};
 
+# the crypted password can be supplied either in the password field
+# OR, alternatively, within a file. this could be useful if putting the crypted
+# password in the profile is undesirable. for this the file will be scanned
+# and the password will be taken from the second field in a colon delimited
+# line, where the first field matches the file_user parameter.
+
+type type_grub_password = {
+    "option"        : string with match (SELF, "^(md5|encrypted)$")
+    "password"      ? string
+    "enabled"       ? boolean
+    "file"          ? string
+    "file_user"     : string = "root"
+};
+
+function validate_grub_password = {
```

Comment:
if you do not reuse this function, just add the verification code to the type as follows
```pan
type type_grub_password = {
  ....
} with {
  # function code
}
```